### PR TITLE
SONAR-9814 make ThreadLocalSettings support DB connectivity issues

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/setting/ThreadLocalSettings.java [hunk 1]
```diff
@@ -26,6 +26,8 @@
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Properties;
+import javax.annotation.CheckForNull;
+import org.apache.ibatis.exceptions.PersistenceException;
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/setting/ThreadLocalSettings.java]: Depend on mybatis is not optimal but it does the job... A unit test could be added by mocking SettingLoader.

File: server/sonar-server/src/main/java/org/sonar/server/setting/ThreadLocalSettings.java [hunk 2]
```diff
@@ -21,11 +21,14 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableMap;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Properties;
+import javax.annotation.CheckForNull;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/setting/ThreadLocalSettings.java]: MINOR Remove this unused import 'javax.annotation.CheckForNull'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/test/java/org/sonar/server/setting/ThreadLocalSettingsTest.java [hunk 3]
```diff
@@ -233,6 +255,89 @@ public void cache_db_calls_if_property_is_not_persisted() {
...
+
+    assertThat(underTest.getProperties())
+      .isEmpty();
+  }
+
+  @Test
+  public void getProperties_return_properties_from_previous_thread_cache_if_DB_error_on_not_first_call() {
+    String key = randomAlphanumeric(3);
+    String value1 = randomAlphanumeric(4);
+    String value2 = randomAlphanumeric(5);
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/setting/ThreadLocalSettingsTest.java]: What's the benefit for using random values ? Test can't be reproduced when it fails.

**Pull request merged**